addons/website_sale/static/src/interactions/sale_variant_mixin.js

```diff
@@ -0,0 +1,619 @@
+     */
+    priceToString(price) {
+        var precision = 2;
+
+        if (document.querySelectorAll(".decimal_precision")) {
+            precision = parseInt(document.querySelector(".decimal_precision").last.dataset.precision);
+        }
+        const formatted = price.toFixed(precision).split(".");
+        const { thousandsSep, decimalPoint, grouping } = localization;
+        formatted[0] = insertThousandsSep(formatted[0], thousandsSep, grouping);
+        return formatted.join(decimalPoint);
+    }
+
+    /**
+     * Returns a throttled `getCombinationInfo` with a leading and a trailing
+     * call, which is memoized per `uniqueId`, and for which previous results
+     * are dropped.
+     *
+     * The uniqueId is needed because on the configurator modal there might be
+     * multiple elements triggering the rpc at the same time, and we need each
+     * individual product rpc to be executed, but only once per individual
+     * product.
+     *
+     * The leading execution is to keep good reactivity on the first call, for
+     * a better user experience. The trailing is because ultimately only the
+     * information about the last selected combination is useful. All
+     * intermediary rpc can be ignored and are therefore best not done at all.
+     *
+     * The keepLast is to make sure we only consider the result of the last call, when several
+     * (asynchronous) calls are done in parallel.
+     *
+     * @param {string} uniqueId
+     * @returns {function}
+    */
+    throttledGetCombinationInfo = memoize(function (self, uniqueId) {
+        const keepLast = new KeepLast();
+        return (ev, params) => keepLast.add(this.throttleForAnimation(this.getCombinationInfo(ev, params)));
+    })
+
+    /**
+     * Toggles the disabled class depending on the parentEl element
+     * and the possibility of the current combination.
+     *
+     * @param {HTMLElement} parentEl
+     * @param {boolean} isCombinationPossible
+    */
+    toggleDisable(parentEl, isCombinationPossible) {
+        parentEl.classList.toggle("css_not_available", !isCombinationPossible);
+    }
+
+    /**
+     * Updates the product image.
+     * This will use the productId if available or will fallback to the productTemplateId.
+     *
+     * @private
+     * @param {HTMLElement} container
+     * @param {boolean} displayImage will hide the image if true. It will use the "invisible" class
+     *   instead of d-none to prevent layout change
+     * @param {integer} product_id
+     * @param {integer} productTemplateId
+    */
+    updateProductImage(container, displayImage, productId, productTemplateId) {
+        var model = productId ? "product.product" : "product.template";
+        var modelId = productId || productTemplateId;
+        var imageUrl = "/web/image/{0}/{1}/" + (this.productImageField ? this.productImageField : "image_1024");
+        var imageSrc = imageUrl.replace("{0}", model).replace("{1}", modelId);
```

Comment:
??? Do we expect `productImageField` to contain `{0}` or `{1}` ? If not, what about a formatted string ?